refactor(l1): remove unwraps from net

ElFantasma opened the pull request:
**Motivation**

`net.rs` module requires some refactors. This one relates to a lot of `unwrap()` methods that should return errors instead 

**Description**

- This makes most of `net.rs` return a `Result<_, NetworkingError>` to be able to use `?` all around.
- Propagates TaskTracker to use on every task spawning (this should fix the graceful shutdown)
- Also refactors the code a bit to make it simpler (eg. removed some `continue` statements that made code harder to follow and refactored some `Optional::unwrap()` as well)

mpaulucci reviewed (approved):
good work!

MarcosNicolau reviewed (approved):
Left some nit comments, other than that it looks very good to me!

fmoletta commented on crates/networking/p2p/net.rs:
```diff
@@ -401,29 +407,29 @@ async fn peers_revalidation(
+                if peer.liveness == 0 {
+                    let new_peer = table.replace_peer(node_id);
+                    if let Some(new_peer) = new_peer {
+                        let ping_hash = ping(
+                            &udp_socket,
+                            udp_addr,
+                            SocketAddr::new(new_peer.node.ip, new_peer.node.udp_port),
+                            &signer,
+                        )
+                        .await?;
```
Im a bit worried about the repercussions of propagating this error. The ping function could fail due to a tcp error if the peer shuts down right? This would propagate the error to `discover_peers` and shut it down as it also propagates the error and we would be left with no discovery service due to a failed ping to one peer

MarcosNicolau replied:
True, we should not propagate these errors, as they will terminate the task. However, we could move the message handling to its own function so that it can return safely without killing the process.

ElFantasma replied:
I'll check it... I guess the previous behavior of panicking wasn't better, right? 🤔

ElFantasma replied:
Ah you are right @fmoletta , it wasn't panicking in the previous version, as the error was prevented. `ping()` shouldn't return error as it is fire and forget

ElFantasma commented:
Ok, reverted most of the `Result<(), NetworkingError>` return type as it turned out all the newly spawned tasks from the TaskTracker could fail and die without any notice. 
Replaced all major failures by `error!()`ing on console.

fmoletta reviewed (approved):
🚀
